File: src/worldpainterStubs/typescript/ScriptingContext.ts
import {World2} from "./World2";

type GetWorldOp = any;
type GetLayerOp = any;
type GetTerrainOp = any;
type ImportHeightMapOp = any;
type ExportWorldOp = any;
type MergeWorldOp = any;
type SaveWorldOp = any;
type GetHeightMapOp = any;
type MappingOp = any;
type InstallCustomTerrainOp = any;
type CreateFilterOp = any;
type GetPlatformOp = any;
type ScriptException = any;
type HeightMap = any;
type MixedMaterial = any;
type Layer = any;

export interface ScriptingContext {
    getVersion(): string;
    getWorld(): GetWorldOp;
    getLayer(): GetLayerOp;
    getTerrain(): GetTerrainOp;
    createWorld(): ImportHeightMapOp;
    exportWorld(world: World2): ExportWorldOp;
    mergeWorld(): MergeWorldOp; // TODO: Define the MergeWorldOp type
    saveWorld(world: World2): SaveWorldOp;
    getHeightMap(): GetHeightMapOp;
    applyHeightMap(heightMap: HeightMap): MappingOp;
    applyLayer(layer: Layer): MappingOp;
    applyTerrain(terrainIndex: number): MappingOp;
    installCustomTerrain(terrain: MixedMaterial): InstallCustomTerrainOp;
    createFilter(): CreateFilterOp;
    getMapFormat(): GetPlatformOp;
    checkGoCalled(commandName: string): void;
}

File: src/worldpainterStubs/java/ScriptingContext.java
package org.pepsoft.worldpainter.tools.scripts;

import org.pepsoft.worldpainter.HeightMap;
import org.pepsoft.worldpainter.MixedMaterial;
import org.pepsoft.worldpainter.World2;
import org.pepsoft.worldpainter.layers.Layer;

public interface ScriptingContextInterface {
    String getVersion();

    GetWorldOp getWorld();

    GetLayerOp getLayer();

    GetTerrainOp getTerrain();

    ImportHeightMapOp createWorld();

    ExportWorldOp exportWorld(World2 world) throws ScriptException;

    // TODO
    MergeWorldOp mergeWorld();

    SaveWorldOp saveWorld(World2 world) throws ScriptException;

    GetHeightMapOp getHeightMap();

    MappingOp applyHeightMap(HeightMap heightMap) throws ScriptException;

    MappingOp applyLayer(Layer layer) throws ScriptException;

    MappingOp applyTerrain(int terrainIndex) throws ScriptException;

    InstallCustomTerrainOp installCustomTerrain(MixedMaterial terrain) throws ScriptException;

    CreateFilterOp createFilter();

    GetPlatformOp getMapFormat();

    void checkGoCalled(String commandName);
}
